File: backend/scripts/createSuperUser.js
/* eslint-disable no-console */
const path = require('path');
const bcrypt = require('bcryptjs');

const db = require(path.join(__dirname, '..', 'src', 'db'));

async function main() {
  const [, , username, password] = process.argv;

  if (!username || !password) {
    console.error('Uso: node scripts/createSuperUser.js <username> <password>');
    process.exit(1);
  }

  try {
    const existing = db.getUserByUsername(username);
    if (existing) {
      console.error(`L'utente "${username}" esiste già.`);
      process.exit(1);
    }

    const passwordHash = await bcrypt.hash(password, 10);
    const user = db.createUser({
      username,
      passwordHash,
      role: 'superuser',
      status: 'approved',
    });
    console.log('Superutente creato con successo:');
    console.log(`  id: ${user.id}`);
    console.log(`  username: ${user.username}`);
  } catch (err) {
    console.error('Errore durante la creazione superutente:', err);
    process.exit(1);
  }
}

main();

File: backend/accounts/management/commands/create_superuser.py
from django.core.management.base import BaseCommand

from accounts.models import User


class Command(BaseCommand):
    help = 'Crea un superutente approvato (equivalente a scripts/createSuperUser.js).'

    def add_arguments(self, parser):
        parser.add_argument('username', type=str)
        parser.add_argument('password', type=str)

    def handle(self, *args, **options):
        username = options['username']
        password = options['password']
        if User.objects.filter(username=username).exists():
            self.stderr.write(self.style.ERROR(f'L\'utente "{username}" esiste già.'))
            return
        User.objects.create_superuser(
            username=username,
            password=password,
            role=User.Role.SUPERUSER,
            status=User.Status.APPROVED,
        )
        self.stdout.write(self.style.SUCCESS('Superutente creato con successo.'))
        self.stdout.write(f'  username: {username}')
